src/tools/rustfmt/src/patterns.rs

```diff
@@ -42,7 +42,7 @@ fn is_short_pattern_inner(pat: &ast::Pat) -> bool {
         | ast::PatKind::Never
         | ast::PatKind::Wild
         | ast::PatKind::Err(_)
-        | ast::PatKind::Lit(_) => true,
+        | ast::PatKind::Expr(_) => true,
```

Comment:
As mentioned above the new `PatKind::Expr` might hold a literal, const block, or path. Can we update this to only return `true` if we know the inner value is a literal. Not sure how we should handle const blocks or paths though.

Reply:
cc: @compiler-errors

Reply:
This is preexisting. No behaviour change towards rustfmt happened in this PR. I'm happy to change things in follow ups, but I want to avoid a behaviour change in this PR

Reply:
@oli-obk I'm also hoping to avoid behavior changes. With the non exhaustive match won't it be easy for behavior changes to accidentally slip in moving forward? To mitigate that I'd like to only return `true` if we know the `PatKind::Expr` contains a literal. Otherwise I think it would be fine to return `false` for now, or is there something that I'm missing?

Reply:
I'd assume returning false would in fact change behaviour, even though likely to the better. This PR just renames the ast variant and does not touch it otherwise. All the changes are on the hir, and even there I do not change where things are, just replacing a large enum with a new one with fewer variants.

Reply:
So for now are these guaranteed to be literals?

Can we add a new arm to the match to make this more explicit (`ast::PatKind::Expr(e) if matches!(e.kind, ExprKind::Lit(_))`)?

Reply:
I'll create a PR tomorrow to make the match exhaustive in rustfmt and panic in the unreachable cases. But today it can be ExprKind::Lit, ExprKind::Unary(Neg, Lit), ConstBlock, Path. This is enforced in the parser iirc